core/src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -378,20 +376,7 @@ public void finalizeRecovery() {
         logger.trace("[{}][{}] finalizing recovery to {}", indexName, shardId, request.targetNode());
 
 
-        cancellableThreads.execute(new Interruptable() {
-            @Override
-            public void run() throws InterruptedException {
-                // Send the FINALIZE request to the target node. The finalize request
-                // clears unreferenced translog files, refreshes the engine now that
-                // new segments are available, and enables garbage collection of
-                // tombstone files. The shard is also moved to the POST_RECOVERY phase
-                // during this time
-                transportService.submitRequest(request.targetNode(), RecoveryTarget.Actions.FINALIZE,
-                        new RecoveryFinalizeRecoveryRequest(request.recoveryId(), request.shardId()),
-                        TransportRequestOptions.builder().withTimeout(recoverySettings.internalActionLongTimeout()).build(),
-                        EmptyTransportResponseHandler.INSTANCE_SAME).txGet();
-            }
-        });
+        cancellableThreads.execute(recoveryTarget::finalizeRecovery);
```

Comment:
The try catch is added because cancellable threads doesn't accept a runnable that throws a checked exception. I don't like it but I opted not to change anything in cancellableThreads . I might give it another go and add an executeWithIOExceptions which allows throwing IO exceptions from  from the runnable.